lib/src/realm_class.dart

```diff
@@ -253,13 +260,21 @@ class Realm {
   /// and [Predicate Programming Guide.](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Predicates/AdditionalChapters/Introduction.html#//apple_ref/doc/uid/TP40001789)
   RealmResults<T> query<T extends RealmObject>(String query, [List<Object> args = const []]) {
     RealmMetadata metadata = _getMetadata(T);
-    final handle = realmCore.queryClass(this, metadata.class_.key, query, args);
-    return RealmResultsInternal.create<T>(handle, this);
+    final queryHandle = realmCore.queryClass(this, metadata.class_.key, query, args);
+    return RealmResultsInternal.create<T>(queryHandle, this);
   }
 
   /// Deletes all [RealmObject]s of type `T` in the `Realm`
   void deleteAll<T extends RealmObject>() => deleteMany(all<T>());
 
+  SubscriptionSet? _subscriptions;
+  /// The active [subscriptions] for this [Realm]
+  SubscriptionSet get subscriptions {
+    if (config is! FlexibleSyncConfiguration) throw RealmError('Does not support flexible synchronization');
+    _subscriptions ??= SubscriptionSetInternal.create(this, realmCore.getSubscriptions(this));
+    return _subscriptions!;
```

Comment:
This needs a mechanics similar to .NET's to refresh the subscription set. Otherwise you will get inconsistent behavior if you open two realm instances:

```dart
final config = Configuration.flexible(...);

final r1 = Realm(config);
print(r1.subscriptions);

final r2 = Realm(config);
r2.subscritpions.update(...);

// Since `_subscriptions` is not null, we will return it even though
// it's pointing to an old version of the subscription set.
print(r1.subscriptions);
```

Reply:
We can obviously refresh on get, but what if users store it in a variable? Then they may never get it again. The safe thing would be to refresh before any operation, but that also seems a bit excessive. 

Anyway, I will add a refresh call here.